staging/src/k8s.io/code-generator/cmd/validation-gen/testscheme/testscheme.go

```diff
@@ -79,3 +91,254 @@ func (s *Scheme) Register(funcs ...func(*Scheme) error) {
 func toRegistrationError(err error) *field.Error {
 	return field.InternalError(nil, fmt.Errorf("registration error: %w", err))
 }
+
+// Test returns a ValidationTestBuilder for this scheme.
+func (s *Scheme) Test(t *testing.T) *ValidationTestBuilder {
+	return &ValidationTestBuilder{t, s}
+}
+
+// ValidationTestBuilder provides convenience functions to build
+// validation tests.
+type ValidationTestBuilder struct {
+	*testing.T
+	s *Scheme
+}
+
+const fixtureEnvVar = "UPDATE_VALIDATION_GEN_FIXTURE_DATA"
+
+// ValidateFixtures ensures that the validation errors of all registered types match what is expected by the test fixture files.
+// For each registered type, a value is created for the type, and populated by fuzzing the value, before validating the type.
+// See ValueFuzzed for details.
+//
+// If the UPDATE_VALIDATION_GEN_FIXTURE_DATA=true environment variable is set, test fixture files are created or overridden.
+//
+// Fixtures:
+//   - validate-false.json: defines a map of registered type to a map of field path to  +validateFalse validations args
+//     that are expected to be returned as errors when the type is validated.
+func (s *ValidationTestBuilder) ValidateFixtures() {
+	flag := os.Getenv(fixtureEnvVar)
+	// Run validation
+	got := map[string]map[string][]string{}
+	for t := range s.s.validationFuncs {
```

Comment:
Currently is the test that is generated deterministic/idempotent?   Not sure if it makes sense to validate or ensure that - not quite sure if it will matter for additional test types.

Reply:
Good question.

For validationFuncs, the order is based on the generated `RegisterValidations` in `zz_generated.validations.go`, which has a consistent order enforced by the generator. 

The fuzzer is a potential sharp edge. It is configured it to use a consistent seed (`NewSource(0)`) to generate data for for each test value.  A new source is created for each value before it is fuzzed so that a value is always fuzzed the same way regardless of what values are fuzzed before it is. 

When fuzzing a single value there is still some room for problems.  If a value has 5 fields, and is fuzzed, and the testdata is stored with the fuzzed data (say, as map keys), then later a field is added in such a way that it is fuzzed before the other 5 fields, the map keys will be different from previously fuzzed map keys in the stored testdata file. Maybe we could do better there? I'm not sure how big of an annoyance it will be in practice.

Reply:
Ah makes sense, I think it's fine as is.